plugins/core-mail/core-mail.go

```diff
@@ -0,0 +1,63 @@
+package core_mail
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	blink_conn "github.com/blinkops/blink-sdk/plugin/connections"
+	log "github.com/sirupsen/logrus"
+	gomail "gopkg.in/mail.v2"
+)
+
+type Plugin struct{}
+
+func (p Plugin) HandleAuth(req *http.Request, conn map[string]string) error {
+	return nil
+}
+
+func (p Plugin) TestConnection(connection *blink_conn.ConnectionInstance) (bool, []byte) {
+	fromEmail, ok := connection.Data["email"]
+	if !ok {
+		return false, []byte("mail connection does not contain an email address")
+	}
+
+	password, ok := connection.Data["password"]
+	if !ok {
+		return false, []byte("mail connection does not contain a password")
+	}
+
+	smtpHost, ok := connection.Data["smtpHost"]
+	if !ok {
+		return false, []byte("mail connection does not contain smtp host server")
+	}
+	smtpPort, ok := connection.Data["smtpPort"]
+	if !ok {
+		return false, []byte("mail connection does not contain smtp host port")
+	}
```

Comment:
Let's change it, we really just want to see that the string isn't empty
```suggestion
	fromEmail := connection.Data["email"]
	if fromEmail != "" {
		return false, []byte("mail connection does not contain an email address")
	}

	password := connection.Data["password"]
	if password != "" {
		return false, []byte("mail connection does not contain a password")
	}

	smtpHost := connection.Data["smtpHost"]
	if smtpHost != "" {
		return false, []byte("mail connection does not contain smtp host server")
	}
	smtpPort := connection.Data["smtpPort"]
	if smtpPort != "" {
		return false, []byte("mail connection does not contain smtp host port")
	}
```